File: youpinhui-sellergoods-interface/src/main/java/com/youpinhui/sellergoods/service/BrandService.java
package com.youpinhui.sellergoods.service;

import java.util.List;
import java.util.Map;

import com.youpinhui.entity.PageResult;
import com.youpinhui.pojo.TbBrand;

/**
 *  provider  Brand interface
 * @author Leon
 *
 */
public interface BrandService {
	
	/**
	 * find all 
	 * @return
	 */
	public List<TbBrand> findAll();

	/**
	 * query according pagination 
	 * @return
	 */
	public PageResult findPage(int page, int size);

	
	/**
	 * Add
	*/
	public void add(TbBrand brand);

	
	/**
	 * update
	 */
	public TbBrand findOne(long id);

	
	/**
	 * get one by id
	 * @param id
	 * @return
	 */
	public void update(TbBrand brand);
	
	
	/**
	 * batch delete
	 * @param ids
	 */
	public void delete(Long[] ids);

	
	/**
	 * 
	 * fuzzy search 
	 *  if the object  is not null , get the parameter from it as the search condition.
	 *
	 * @param pageNum current page number
	 * @param pageSize size of the page to display
	 * @return
	 */
	public PageResult findPage(TbBrand brand, int page, int size);
	
	
	/**
	 * 
	 * return select2 dropdown menu data 
	 */
	public List<Map> selectOptionList();
	
}

File: youpinhui-manager-web/src/main/webapp/js/service/brandService.js
app.service("brandService",function($http){
				
    this.findAllBrand=function(){
        return $http.get("../brand/findAll.do");
    }
    this.findPage=function(page,size){
        return $http.get('../brand/findPage.do?page='+page+'&size='+size);
    }
    this.add=function(entity){
        return $http.post('../brand/add.do',entity);
    }

    this.findOne=function(id){
        return  $http.get('../brand/findOne.do?id='+id);
    }
   

})